packages/metro/src/DeltaBundler/traverseDependencies.js

```diff
@@ -323,9 +385,17 @@ function removeDependency<T>(
 
   module.inverseDependencies.delete(parentModule.path);
 
-  // This module is still used by another modules, so we cannot remove it from
-  // the bundle.
-  if (module.inverseDependencies.size) {
+  // Even if there are modules still using parentModule, we want to ensure
+  // there isn't circular dependency. Thus, we check if it can be safely remove
+  // by tracing back the inverseDependencies.
+  if (
+    module.inverseDependencies.size &&
```

Comment:
Can do the `module.inverseDependencies.size` check in `canSafelyRemoveFromParentModule`

Reply:
I think this may make it clear on what `canSafelyRemoveFromParentModule` function intends to do, as way to signify that if `module.inverseDependencies.size` is non zero then we need additional check for all of the inverse dependencies.